### PR TITLE
Update workflows for Xcode 14

### PR DESCRIPTION
**Pull request by douglowder**
# Why

Fix CI build issues

# How

Update workflows

# Test Plan

CI should pass

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: .github/workflows/client-ios.yml [hunk 1]
```diff
@@ -42,14 +42,14 @@ concurrency:
...
+    runs-on: macos-12
     steps:
       - name: 👀 Checkout
         uses: actions/checkout@v3
         with:
           submodules: true
-      - name: 🔨 Switch to Xcode 13.2.1
-        run: sudo xcode-select --switch /Applications/Xcode_13.2.1.app
+      - name: 🔨 Switch to Xcode 14.0
+        run: sudo xcode-select --switch /Applications/Xcode_14.0.app
```
Comment (tsapeta) [inline .github/workflows/client-ios.yml:52]: Why not use the latest available 14.1?
```suggestion
        run: sudo xcode-select --switch /Applications/Xcode_14.1.app
```

**Review by tsapeta — CHANGES_REQUESTED** [review]

File: .github/workflows/client-ios.yml (hunk 1 above)
Comment (douglowder) [reply .github/workflows/client-ios.yml:52]: It wasn't clear if GitHub supported 14.1 yet but I will try that next.

Comment (tsapeta) [reply .github/workflows/client-ios.yml:52]: They do, see https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md

Comment (douglowder) [reply .github/workflows/client-ios.yml]: Done

**Review by tsapeta — APPROVED** [review]

**Pull request merged**